Server/Classes/Handlers/VehicleSpawner.sqx

```diff
@@ -125,12 +125,69 @@ namespace Intrusion.Server
 		    clearBackpackCargoGlobal _vehicleObj;
 		    clearItemCargoGlobal _vehicleObj;
 		    
-		    {
-            	_vehicleObj addBackpackCargoGlobal [_x, 1];
-		    } foreach _vehicleClassInfo.CargoWeapons as String;
-		    
         	_intrusionVehicle = [_vehicleObj, _side, _professionType, _group, _cost] new IntrusionVehicle;
         	_vehicleObj setVariable ["IntrusionVehicle", _intrusionVehicle, true];
+        	
+	    	{
+	    		private ["_attachObj" as Object, "_attachObjInfo" as VehicleAttachmentInfo];
+	    		_attachObjInfo = _x;
+	    		
+	    		_attachObj = _attachObjInfo.ClassName createVehicle (getPos _vehicleObj);
+				_attachObj attachTo [_vehicleObj, _attachObjInfo.AttachOffset];
+				clearWeaponCargoGlobal _attachObj;
+				clearItemCargoGlobal _attachObj;
+				clearMagazineCargoGlobal _attachObj;
+				
+				// Fill the weaponbox again
+				{
+	            	if (_x.ClassName isKindOf ["Bag_Base", configFile>>"CfgVehicles"]) then {
+			    		_vehicleObj addBackpackCargoGlobal [_x.ClassName, _x.Quantity];
+			    	} else {
+	    	        	_vehicleObj addItemCargoGlobal [_x.ClassName, _x.Quantity];
+	    	        };
+			    } foreach _vehicleClassInfo.CargoItems as VehicleCargoInfo;
+			    
+			    // When this is a part of the vehicle, add the IntrusionVehicle object.
+				// This will lock the crate to right profession.
+			    _attachObj setVariable ["IntrusionVehicle", _intrusionVehicle, true];
+			    
+			    // When vehicle explodes, destroy ammo box
+				_vehicleObj addEventHandler ["Killed", {
+					params ["_veh"];
+					if (!canMove _veh) then {
+						_veh removeEventHandler ["Killed", _thisEventHandler];
+						{
+							[_x] spawn {
+								params["_obj"];
+								_obj setDamage 1;
+								sleep 3;
+								deleteVehicle _obj;
+							};
+						} forEach attachedObjects _veh;
+					};
+				}];
+				
+				// When vehicle is removed, delete the attachments
+				_vehicleObj addEventHandler ["Deleted", {
+					params ["_veh"];
+					{
+						deleteVehicle _x;
+					} forEach attachedObjects _veh
+				}];
+			    
+	    	} foreach _vehicleClassInfo.Attachments as VehicleAttachmentInfo;
+		    
+		    // If there are no attachments to this vehicle add cargo items to vehicle cargo .
+		    if (count _vehicleClassInfo.Attachments == 0) then {
```

Comment:
So you assume that an attachment is always a container? Maybe one should here also test if the attachment is an attachment that can contain stuff. Or alternatively, test if the vehicle has its own cargo space or not.

(If we assume that an attachment is always a cargo space, lets change the name of VehicleAttachmentInfo to VehicleCargoAttachmentInfo.)